### PR TITLE
added malloc checks

### PR DESCRIPTION
**Pull request by ericwol-msft**

### CODE REVIEW

File: src/tpm_socket_comm.c [hunk 1]
```diff
@@ -57,17 +58,19 @@ static int add_to_buffer(TPM_SOCKET_INFO* socket_info, const unsigned char* byte
     }
     else
     {
-        new_buff = (unsigned char*)realloc(socket_info->recv_bytes, socket_info->recv_length + length);
-}
+        size_t realloc_size = safe_add_size_t(socket_info->recv_length, length);
+        new_buff = (unsigned char*)realloc(socket_info->recv_bytes, realloc_size);
```
Comment (vaavva) [inline src/tpm_socket_comm.c:62]: Do we need to check that realloc_size isn't SIZE_MAX before the realloc?

File: src/tpm_socket_comm.c [hunk 2]
```diff
@@ -57,17 +58,19 @@ static int add_to_buffer(TPM_SOCKET_INFO* socket_info, const unsigned char* byte
...
     if (new_buff == NULL)
     {
         result = MU_FAILURE;
     }
     else
     {
         socket_info->recv_bytes = new_buff;
-        memcpy(socket_info->recv_bytes + socket_info->recv_length, bytes, length);
-        socket_info->recv_length += length;
+        memcpy(safe_add_size_t(socket_info->recv_bytes, socket_info->recv_length), bytes, length);
```
Comment (vaavva) [inline src/tpm_socket_comm.c:72]: same here, does the return value of safe_add_size_t need to be checked?

File: src/tpm_socket_comm.c [hunk 3]
```diff
@@ -83,8 +86,9 @@ static void remove_from_buffer(TPM_SOCKET_INFO* socket_info, size_t length)
     }
     else
     {
-        unsigned char* new_buff = (unsigned char*)malloc(socket_info->recv_length - length);
-        memcpy(new_buff, &socket_info->recv_bytes[length], socket_info->recv_length - length);
+        size_t malloc_size = safe_subtract_size_t(socket_info->recv_length, length);
+        unsigned char* new_buff = (unsigned char*)malloc(malloc_size);
```
Comment (ewertons) [inline src/tpm_socket_comm.c:90]: Same here, we should check malloc_size for zero. There was another ticket opened for trying to malloc with zero, which could be result in unpredictable behavior depending on the system.

**Review by ewertons — APPROVED** [review]
Approved with requests for change.

File: src/tpm_socket_comm.c (hunk 2 above)
Comment (ericwol-msft) [reply src/tpm_socket_comm.c]: reverted

File: src/tpm_socket_comm.c [hunk 4]
```diff
@@ -83,11 +102,15 @@ static void remove_from_buffer(TPM_SOCKET_INFO* socket_info, size_t length)
...
-        socket_info->recv_bytes = new_buff;
-        socket_info->recv_length -= length;
+        size_t malloc_size = safe_subtract_size_t(socket_info->recv_length, length);
+        if (malloc_size != SIZE_MAX)
+        {
+            unsigned char* new_buff = (unsigned char*)malloc(malloc_size);
+            memcpy(new_buff, &socket_info->recv_bytes[length], malloc_size);
+            free(socket_info->recv_bytes);
+            socket_info->recv_bytes = new_buff;
+            socket_info->recv_length -= length;
```
Comment (vaavva) [inline src/tpm_socket_comm.c:112]:
```suggestion
            socket_info->recv_length = malloc_size;
```

**Review by vaavva — APPROVED** [review]

**Pull request merged**